modules/internal/SharedObject.chpl

```diff
@@ -435,7 +435,7 @@ module SharedObject {
       if _to_nilable(chpl_t) == chpl_t {
         return chpl_p;
       } else {
-        return chpl_p!;
+        return _to_nonnil(chpl_p);
```

Comment:
This change seems curious... Not sure we want to change this to no longer halt if the non-nilable shared contains nil. Maybe whatever reason you had for adding it could use `chpl_p` directly or some other method?

Reply:
I was actually super confused that there are cases where a non-nilable would do `postfix!`. I had a similar confusion about why the compiler doesn't automatically optimize `someNonNilable == nil` to evaluate to `true`. Actually wait. We might need `postfix!` for runtime nil checking on e.g. `owned`.

I can try to use `chpl_p` directly for the checking, thanks!

Reply:
That's fair, and arguably we could elide this check unless `--checks` is on, but for now let's leave it alone.